### PR TITLE
Expose new HIP-786 staking properties

### LINKED ISSUE #6792: HIP-786: Expose new staking properties
### Problem

[HIP-786](https://hips.hedera.com/hip/hip-786) proposes that consensus nodes add 5 new fields and deprecate one field from the `NodeStakeTransaction`.

### Solution

* Add `reserved_staking_rewards`, `unreserved_staking_reward_balance`, `reward_balance_threshold`, `max_stake_rewarded`, `max_total_reward` as new fields to `/api/v1/network/stake`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jascks**
**Description**:

[HIP-786](https://hips.hedera.com/hip/hip-786) proposes that consensus nodes add 5 new fields and deprecates one field from the `NodeStakeTransaction`.

* Added `reserved_staking_rewards`, `unreserved_staking_reward_balance`, `reward_balance_threshold`, `max_stake_rewarded`, `max_total_reward` as new fields to `/api/v1/network/stake` and OpenAPI spec.
* retained `staking_reward_rate` for backward compatibility.

**Related issue(s)**:

Fixes #6792

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 1]
```diff
@@ -2807,9 +2836,19 @@ components:
           description: The minimum balance of staking reward account 0.0.800 required to active rewards
           type: integer
           format: int64
+        unreserved_staking_reward_balance:
+          description:
+            The unreserved balance of account 0.0.800 at the close of the just-ending period;
+            this value is used to compute the HIP-782 balance ratio
+          format: int64
+          type: integer
       example:
```
Comment (jascks) [inline hedera-mirror-rest/api/v1/openapi.yml:2845]: I have no idea what would make more reasonable example values for the new fields. Suggestions welcomed!

File: hedera-mirror-rest/api/v1/openapi.yml [hunk 2]
```diff
@@ -2761,24 +2761,53 @@ components:
...
         - staking_period
         - staking_period_duration
         - staking_periods_stored
         - staking_reward_fee_fraction
         - staking_reward_rate
         - staking_start_threshold
+        - unreserved_staking_reward_balance
       properties:
+        max_stake_rewarded:
+          description:
```
Comment (xin-hedera) [inline hedera-mirror-rest/api/v1/openapi.yml:2780]: for multi-line string

```suggestion
          description: |
```

Comment (jascks) [reply hedera-mirror-rest/api/v1/openapi.yml]: Ah, thank you for catching that. Done.

File: hedera-mirror-rest/__tests__/specs/network/stake/no-params.json
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/network/stake/no-params.json:1]: Need to add fields to `integrationDomainOps.js`.

Comment (jascks) [reply hedera-mirror-rest/__tests__/specs/network/stake/no-params.json:1]: They are there. I was forced to do so in my previous PR.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**